AffiliateWP/affiliatewp-checkout-referrals
# issue/60 - Get tracking cookie name from cookie name getter

Issue #60


**Housekeeping**


**Proposed changes**


The hard-coded tracking cookie on this add-on is only used to check if an affiliate link is already being tracked so that the affiliate selection dropdown is not shown.

## Before changes - cart page from an affiliate link on a Pantheon's edge network
<img width="1162" alt="imagem" src="https://user-images.githubusercontent.com/498472/122479235-9f691080-cfc2-11eb-9dfc-c8eadcc3511f.png">
(The affiliate selection dropdown is still visible, tracking cookie not detected)

## After changes - cart page from an affiliate link on a Pantheon's edge network
<img width="1162" alt="imagem" src="https://user-images.githubusercontent.com/498472/122479173-8c564080-cfc2-11eb-810b-e5bd689d2d62.png">
(The affiliate selection dropdown not shown, tracking cookie detected)

## How to test it
1. Use the AffiliateWP branch `issue/3937`
2. Install and activate Checkout Referalls add-on
3. On `/includes/class-affiliatewp-affiliate-landing-pages.php:202` force AffiliateWP version check to be 2.7.1+: `$affwp_version = '2.7.1';` so that it doesn't depend on AffiliateWP version
3. On wp-config add the following server environment variable: `$_ENV['PANTHEON_ENVIRONMENT'] = 1;` to simulate being on Pantheon's edge network
5. Visit the site with an affiliate link and confirm the tracking cookies are prefixed with `wp_`
6. Add a product to the cart and confirm the affiliate selection dropdown doesn't show on the checkout page - this confirms that the tracking cookie was correctly identified by the add-on
